### ISSUE TITLE
SIGSEGV (Segmentation Fault) when running train_stage1.sh

### ISSUE THREAD

**Issue by zxcbb16** [soda-inria/tabicl#30]
#### Describe the bug
When executing the `scripts/train_stage1.sh` script for TabICL Stage 1 training, the program consistently crashes with a **SIGSEGV (Segmentation Fault, exitcode: -11)** error, regardless of using single/multi GPU training. I would greatly appreciate any guidance or suggestions to resolve this issue.

#### Error Log (Key Snippets)
DDP training with 4 processes
Per-GPU batch size: 128
Model has 27051658 parameters.
E0114 12:30:35.763578 26104 site-packages/torch/distributed/elastic/multiprocessing/api.py:874] failed (exitcode: -11) local_rank: 0 (pid: 26215)
torch.distributed.elastic.multiprocessing.errors.ChildFailedError:
Root Cause:
[0]:
rank : 0 (local_rank: 0)
exitcode : -11 (pid: 26215)
traceback : Signal 11 (SIGSEGV) received by PID 26215

#### Steps to Reproduce
1. Install the project in editable mode: `cd tabicl; pip install -e .`
2. Execute the training script: `bash scripts/train_stage1.sh`
3. The program crashes immediately after model parameter initialization (shows "Model has 27051658 parameters.")

#### Expected Behavior
The training script should run without segmentation faults, starting normal training after model initialization.

#### Additional Context
The error occurs right after "Model has 27051658 parameters." is printed, suggesting the crash is related to model forward pass initialization, DDP gradient sync, or custom modules (e.g., RoPE encoding, row/column Transformer) in TabICL.

Thank you very much for your time and assistance in looking into this issue!
